Support sending signals

**Description**

Signal events are introduced in camunda 8.3.0. When using the zeebe client to create a new signal event in a @ZeebeProcessTest the following error occurs:

```
io.camunda.zeebe.client.api.command.ClientStatusException
 at io.camunda.zeebe.client.impl.ZeebeClientFutureImpl.transformExecutionException([ZeebeClientFutureImpl.java:116](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html))
 at io.camunda.zeebe.client.impl.ZeebeClientFutureImpl.join([ZeebeClientFutureImpl.java:54](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html))
...
Caused by: java.util.concurrent.ExecutionException: io.grpc.StatusRuntimeException: UNKNOWN
 at java.base/java.util.concurrent.CompletableFuture.reportGet([CompletableFuture.java:396](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html))
 at java.base/java.util.concurrent.CompletableFuture.get([CompletableFuture.java:2073](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html))
 at io.camunda.zeebe.client.impl.ZeebeClientFutureImpl.join([ZeebeClientFutureImpl.java:52](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html))
 ... 72 more
Caused by: io.grpc.StatusRuntimeException: UNKNOWN
 at io.grpc.Status.asRuntimeException([Status.java:537](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html))
 at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose([ClientCalls.java:481](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html))
...
```

This issue is for signal events to be supported in zeebe-process-test

--- a/engine/src/main/java/io/camunda/zeebe/process/test/engine/GrpcResponseMapper.java
+++ b/engine/src/main/java/io/camunda/zeebe/process/test/engine/GrpcResponseMapper.java
@@ -14,6 +14,8 @@ import com.google.rpc.Status;
 import io.camunda.zeebe.gateway.protocol.GatewayOuterClass.ActivateJobsRequest;
 import io.camunda.zeebe.gateway.protocol.GatewayOuterClass.ActivateJobsResponse;
 import io.camunda.zeebe.gateway.protocol.GatewayOuterClass.ActivatedJob;
+import io.camunda.zeebe.gateway.protocol.GatewayOuterClass.BroadcastSignalRequest;
+import io.camunda.zeebe.gateway.protocol.GatewayOuterClass.BroadcastSignalResponse;
 import io.camunda.zeebe.gateway.protocol.GatewayOuterClass.CancelProcessInstanceRequest;
 import io.camunda.zeebe.gateway.protocol.GatewayOuterClass.CancelProcessInstanceResponse;
 import io.camunda.zeebe.gateway.protocol.GatewayOuterClass.CompleteJobRequest;
@@ -59,6 +61,7 @@ import io.camunda.zeebe.protocol.impl.record.value.job.JobBatchRecord;
 import io.camunda.zeebe.protocol.impl.record.value.job.JobRecord;
 import io.camunda.zeebe.protocol.impl.record.value.processinstance.ProcessInstanceCreationRecord;
 import io.camunda.zeebe.protocol.impl.record.value.processinstance.ProcessInstanceResultRecord;
+import io.camunda.zeebe.protocol.impl.record.value.signal.SignalRecord;
 import io.camunda.zeebe.protocol.impl.record.value.variable.VariableDocumentRecord;
 import io.camunda.zeebe.protocol.record.RejectionType;
 import io.camunda.zeebe.protocol.record.intent.Intent;
@@ -93,7 +96,8 @@ class GrpcResponseMapper {
           Map.entry(ResolveIncidentRequest.class, this::createResolveIncidentResponse),
           Map.entry(SetVariablesRequest.class, this::createSetVariablesResponse),
           Map.entry(UpdateJobRetriesRequest.class, this::createJobUpdateRetriesResponse),
-          Map.entry(ModifyProcessInstanceRequest.class, this::createModifyProcessInstanceResponse));
+          Map.entry(ModifyProcessInstanceRequest.class, this::createModifyProcessInstanceResponse),
+          Map.entry(BroadcastSignalRequest.class, this::createBroadcastSignalResponse));
 
   GeneratedMessageV3 map(
       final Class<? extends GeneratedMessageV3> requestType,
@@ -285,6 +289,13 @@ class GrpcResponseMapper {
 
   private GeneratedMessageV3 createModifyProcessInstanceResponse() {
     return ModifyProcessInstanceResponse.newBuilder().build();
+  }
+
+  private GeneratedMessageV3 createBroadcastSignalResponse() {
+    final SignalRecord signal = new SignalRecord();
+    signal.wrap(valueBufferView);
+
+    return BroadcastSignalResponse.newBuilder().setKey(key).build();
   }
 
   private GeneratedMessageV3 createResolveIncidentResponse() {

--- a/engine/src/main/java/io/camunda/zeebe/process/test/engine/GrpcToLogStreamGateway.java
+++ b/engine/src/main/java/io/camunda/zeebe/process/test/engine/GrpcToLogStreamGateway.java
@@ -493,11 +493,15 @@ class GrpcToLogStreamGateway extends GatewayGrpc.GatewayImplBase {
     final var requestId =
         gatewayRequestStore.registerNewRequest(request.getClass(), responseObserver);
 
+    final SignalRecord command = new SignalRecord().setSignalName(request.getSignalName());
+
+    if (!request.getVariables().isEmpty()) {
+      command.setVariables(
+          BufferUtil.wrapArray(MsgPackConverter.convertToMsgPack(request.getVariables())));
+    }
+
     writer.writeCommandWithoutKey(
-        new SignalRecord()
-            .setSignalName(request.getSignalName())
-            .setVariables(
-                BufferUtil.wrapArray(MsgPackConverter.convertToMsgPack(request.getVariables()))),
+        command,
         prepareRecordMetadata()
             .requestId(requestId)
             .valueType(ValueType.SIGNAL)

--- a/engine/src/test/java/io/camunda/zeebe/process/test/engine/EngineClientTest.java
+++ b/engine/src/test/java/io/camunda/zeebe/process/test/engine/EngineClientTest.java
@@ -15,6 +15,7 @@ import io.camunda.zeebe.client.api.ZeebeFuture;
 import io.camunda.zeebe.client.api.command.ClientException;
 import io.camunda.zeebe.client.api.response.ActivateJobsResponse;
 import io.camunda.zeebe.client.api.response.ActivatedJob;
+import io.camunda.zeebe.client.api.response.BroadcastSignalResponse;
 import io.camunda.zeebe.client.api.response.BrokerInfo;
 import io.camunda.zeebe.client.api.response.DeploymentEvent;
 import io.camunda.zeebe.client.api.response.EvaluateDecisionResponse;
@@ -1012,5 +1013,56 @@ class EngineClientTest {
     assertThat(form.getResourceName()).isEqualTo("form/test-form-1.form");
     assertThat(form.getFormKey()).isPositive();
     assertThat(form.getFormId()).isEqualTo("Form_0w7r08e");
+  }
+
+  @Test
+  void shouldBroadcastSignal() {
+    // given
+    zeebeClient
+        .newDeployResourceCommand()
+        .addProcessModel(
+            Bpmn.createExecutableProcess("simpleProcess")
+                .startEvent()
+                .signal("signal")
+                .endEvent()
+                .done(),
+            "simpleProcess.bpmn")
+        .send()
+        .join();
+
+    // when
+    final BroadcastSignalResponse response =
+        zeebeClient.newBroadcastSignalCommand().signalName("signal").send().join();
+
+    // then
+    assertThat(response.getKey()).isPositive();
+  }
+
+  @Test
+  void shouldBroadcastSignalWithVariables() {
+    // given
+    zeebeClient
+        .newDeployResourceCommand()
+        .addProcessModel(
+            Bpmn.createExecutableProcess("simpleProcess")
+                .startEvent()
+                .signal("signal")
+                .endEvent()
+                .done(),
+            "simpleProcess.bpmn")
+        .send()
+        .join();
+
+    // when
+    final BroadcastSignalResponse response =
+        zeebeClient
+            .newBroadcastSignalCommand()
+            .signalName("signal")
+            .variable("foo", "bar")
+            .send()
+            .join();
+
+    // then
+    assertThat(response.getKey()).isPositive();
   }
 }

